### PR TITLE
(even more) Dashboard fixes

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Implements https://github.com/enso-org/cloud-v2/issues/484
- Mostly implements https://github.com/enso-org/cloud-v2/issues/481 - ~~the one that uses offline mode is *not* implemented, as offline mode is not yet merged in~~
- Fixes some of https://github.com/enso-org/cloud-v2/issues/483. Most are already being fixed in https://github.com/enso-org/enso/pull/6728; the project create form one is being added as a separate PR because getting a styled dropdown will be a bit complicated. What is being implemented here is:
  - removing modal from main upload file icon
  - reducing delay for a new project to get added to the UI

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

**Comment by PabloBuchu** [comment]
Can we do something with this ugly row resizing when adding new project?

https://github.com/enso-org/enso/assets/2855109/979b53bb-a839-497b-8a93-661ca8ac16b3

**Comment by somebody1234** [comment]
everything seems to work, some bugs were fixed. unfortunately couldn't test whether you can do actions (create, rename, delete projects) in offline mode via `-authentication=false`

**Comment by somebody1234** [comment]
also note that in offline mode the user menu is completely disabled. this is intentional as i'm assuming that `authentication=false` means they don't ever want to see the authentication flow

**Comment by somebody1234** [comment]
oh also i haven't tested whether `-authentication=false` works when you're logged out in the regular dashboard

**Comment by somebody1234** [comment]
oh, it looked like it was freezing because graal was downloading

**Comment by PabloBuchu** [comment]
@somebody1234 please fix conflicts and fix linter

**Comment by PabloBuchu** [comment]
@somebody1234  `-authentication=false` does not have a switcher (it should be normal offline mode) and templates looks weird now

Screenshot 2023-06-08 at 11 53 17

**Review by PabloBuchu — APPROVED** [review]

**Comment by somebody1234** [comment]
- switcher for `-authentication=false`: reintroduced
  - logins: *should* work - tested github, google and password, *but* not the login flow as that's difficult to test without deep link support
- templates: reverted to occupy the full width

**Comment by somebody1234** [comment]
note: the login screen flashes when you're logging in from `-authentication=false`. i tried showing a toast message but it turns out that sometimes overlaps with the sign in toast message... not sure what should be done there.
also:

- setting username: works
- create new account flow: does not redirect back to login page :warning: (edit: now fixed)

**Comment by sylwiabr** [comment]
@indiv0 can you QA this?

**Comment by indiv0** [comment]
@sylwiabr QA in progress, working through the prerequisite Enso issues in order.

**Comment by indiv0** [comment]
~~**TL;DR: Multiple issues found, QA failed ❌**~~

**QA being repeated, currently in progress**

# Scenario Descriptions

<table style="border: 1px solid black;">
    <thead>
        <tr>
            <th style="border: 1px solid black;" rowspan=4>Scenario</th>
        </tr>
        <tr>
            <th style="border: 1px solid black;" colspan=5>Command</th>
            <th style="border: 1px solid black;" colspan=2>Features</th>
            <th style="border: 1px solid black;" rowspan=3>Debug Scene</th>
        </tr>
        <tr style="border: 1px solid black;">
            <th style="border: 1px solid black;">GUI</th>
            <th style="border: 1px solid black;" colspan=2>IDE</th>
            <th style="border: 1px solid black;" colspan=2>NPM</th>
            <th style="border: 1px solid black;" rowspan=2>Auth</th>
            <th style="border: 1px solid black;" rowspan=2>New Dashboard</th>
        </tr>
        <tr style="border: 1px solid black;">
            <th style="border: 1px solid black;">Watch</th>
            <th style="border: 1px solid black;">Watch</th>
            <th style="border: 1px solid black;">Build</th>
            <th style="border: 1px solid black;">build-dashboard</th>
            <th style="border: 1px solid black;">watch-dashboard</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td style="border: 1px solid black;">0</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">❌</td>
            <td style="border: 1px solid black;">❌</td>
            <td style="border: 1px solid black;">❌</td>
        </tr>
        <tr>
            <td style="border: 1px solid black;">1</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">✅</td>
        </tr>
        <tr>
            <td style="border: 1px solid black;">2</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">❌</td>
        </tr>
        <tr>
            <td style="border: 1px solid black;">3</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">❌</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">❌</td>
        </tr>
        <tr>
            <td style="border: 1px solid black;">4</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">❌</td>
            <td style="border: 1px solid black;">❌</td>
        </tr>
        <tr>
            <td style="border: 1px solid black;">5</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">❌</td>
            <td style="border: 1px solid black;">❌</td>
            <td style="border: 1px solid black;">❌</td>
        </tr>
        <tr>
            <td style="border: 1px solid black;">6</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">❌</td>
        </tr>
        <tr>
            <td style="border: 1px solid black;">7</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
        </tr>
        <tr>
            <td style="border: 1px solid black;">8</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
        </tr>
    </tbody>
</table>

# Scenario 0

```sh
./run gui watch \
    --skip-wasm-opt \
    --wasm-profile=dev \
    --skip-version-check \
    --profiling-level=objective \
    --wasm-log-level=info \
    --wasm-uncollapsed-log-level=info
```

- ✅ GUI at http://localhost:8080 opens
- ✅ Can login
- ✅ Dashboard shows up
- ✅ Nodes in Cloud project are visible
- ✅ IDE rebuilds for edits in `app/gui/**/*.rs`
- ✅ IDE reloads for edits in `app/gui/**/*.rs`
- ✅ IDE rebuilds for edits in `app/ide-desktop/lib/dashboard/src/authentication/src/**/*.tsx`
- ✅ IDE reloads for edits in `app/ide-desktop/lib/dashboard/src/authentication/src/**/*.tsx`

<details>
<summary>
Screenshot
</summary>
image
</details>


# Scenario 1

```sh
ENSO_BUILD_TARGET=dir \
    ./run ide watch \
    --skip-wasm-opt \
    --wasm-profile=dev \
    --backend-source=release \
    --backend-release=nightly \
    --skip-version-check \
    --profiling-level=objective \
    --wasm-log-level=info \
    --wasm-uncollapsed-log-level=info \
    --ide-option=-feature-preview.new-dashboard \
    --ide-option=-debug.dev-tools \
    --ide-option=-debug \
    --ide-option=--inspect=5858 \
    --ide-option=-authentication \
    --ide-option=-startup.entry=text_area
```

- ✅ Debug scene is visible
- ✅ Logs appear with `console.log("Hello")`

<details>
<summary>
Screenshot
</summary>
image

</details>

# Scenario 2

```sh
ENSO_BUILD_TARGET=dir \
    ./run ide watch \
    --skip-wasm-opt \
    --wasm-profile=dev \
    --backend-source=release \
    --backend-release=nightly \
    --skip-version-check \
    --profiling-level=objective \
    --wasm-log-level=info \
    --wasm-uncollapsed-log-level=info \
    --ide-option=-feature-preview.new-dashboard \
    --ide-option=-debug.dev-tools \
    --ide-option=-debug \
    --ide-option=--inspect=5858 \
    --ide-option=-authentication
```

- ✅ Nodes in Cloud project are visible
- ⚠️ (Re-testing) Error in logs when re-opening existing and currently open Cloud Project, `Content-Length` is not provided (see screenshots)
    - Unable to reproduce (c.f., https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268)
- ⚠️ (Re-testing) `checkResources()` gets stuck on `400 Bad Request` for an extremely long time
- ✅ Logs appear with `console.log("Hello")`
- ✅ S3 CDN bucket is no longer `https://ensocdn.s3.us-west-1.amazonaws.com`, and is instead `https://cdn.enso.org`
- ⚠️ (Re-testing) Overlap in text for file rename modal & close button for file rename modal (see screenshots)
	- Not necessarily caused by this PR, but worth addressing
	- Marked as now fixed, per https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268
- ✅ When a project is opened in the background, the inputs (e.g., searcher, or add new directory) work. Things can be written, per [this issue](https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099).
	- Known issue, potentially because of a keyboard event handler that the IDE is registering, per [this discussion](https://github.com/enso-org/cloud-v2/issues/483#issuecomment-1571907519)
- ⚠️ (Re-testing) When a project is opened in the background, when you switch between modes, the `stop` icon should not be animated (c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099)
	- ~~Marked as "might have been fixed" in https://github.com/enso-org/enso/pull/6728~~
	- ~~Does not appear to be fixed (see screencasts)~~
	- Marked as now fixed, per https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268
- ⚠️ (Re-testing) `+` should appear next to the project; that button should create a new project from empty template
	- c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099
	- Marked as TODO per [this comment](https://github.com/enso-org/cloud-v2/issues/483#issuecomment-1571907519)
	- Tracked as https://github.com/enso-org/cloud-v2/issues/480
	- Marked as implemented in https://github.com/enso-org/enso/pull/6923
	- Icon has been added and works as described, but a duplicate "creating a new project" toast appears when it is used (see screencasts)
	- Duplicate toast is marked as fixed in a later PR (c.f., https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268)
- ✅ When starting a cloud project, a toast saying "We are preparing an environment for you. Please be patient." should appear (c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099)
	- Originally reported in https://github.com/enso-org/cloud-v2/issues/449
	- Supposed to be fixed by [Save backend type in localStorage, and automatically open projects when ready enso#6728](https://github.com/enso-org/enso/pull/6728)
- ✅ When clicking on a new empty project or one of the project templates, the project should be created and immediately started (c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099)
	- Supposed to be fixed by [Save backend type in localStorage, and automatically open projects when ready enso#6728](https://github.com/enso-org/enso/pull/6728)
- ⚠️ (Re-testing) When clicking the main upload file icon, no modal should appear. A finder should appear directly. The file should be uploaded into current directory without any more confirmations (c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099)
	- Marked as TODO per [this comment](https://github.com/enso-org/cloud-v2/issues/483#issuecomment-1571907519)
	- Marked as implemented in [this PR](https://github.com/enso-org/enso/pull/6919) per [this comment](https://github.com/enso-org/enso/pull/6919#issue-1736336292)
	- Not fixed; modal still appears (see screenshots)
		- Potentially because fix hasn't been merged into this branch?
        - Marked as now fixed, per https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268
- ✅ ~~Project create form is added~~
	- ~~**TODO**: Doesn't appear to be tracked as an issue anywhere?~~
	- ~~Being added as a separate PR per [this comment](https://github.com/enso-org/enso/pull/6919#issue-1736336292)~~
	- Marked as no longer planned, per https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268
     > it was changed to create a project directly - so there is no issue here
- ✅ There should be no lag between adding a new project and it appearing in the projects list (previously there was a 0.5-1s delay)
	- c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099
	- Marked as TODO per [this comment](https://github.com/enso-org/cloud-v2/issues/483#issuecomment-1571907519):
     > can do, i think the best solution here is to manually prepend the new project, but also `doRefresh()` to ensure correctness
	- Marked as implemented in [this PR](https://github.com/enso-org/enso/pull/6919) per [this comment](https://github.com/enso-org/enso/pull/6919#issue-1736336292)
- ⚠️ (Re-testing) Click avatar -> select change password -> click outside, hide modal -> click avatar (doesn't work) -> click avatar (works)
	- c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099
	- Supposed to be fixed by [Save backend type in localStorage, and automatically open projects when ready enso#6728](https://github.com/enso-org/enso/pull/6728)
	- Not fixed; no "change password" option appears under the avatar menu at all (see screenshots)
	- No "change password" option is intentional on GitHub and Google logins, as they do not have a password (c.f., https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268)
- ✅ Ugly row resizing (c.f., https://github.com/enso-org/enso/pull/6919#issuecomment-1573413827) does not occur when adding a new project

- ✅ IDE rebuilds for edits in `app/gui/**/*.rs`
- ✅ IDE rebuilds for edits in `app/ide-desktop/lib/dashboard/src/authentication/src/**/*.tsx`

<details>
<summary>
Screenshot(s)/screencast(s)
</summary>
image
image
image
image
image
<video src="https://github.com/enso-org/enso/assets/641547/a01440a6-3944-4f61-985b-8ef87343d612" autoplay controls loop>
</video>
<video src="https://github.com/enso-org/enso/assets/641547/ae3581b3-c8fe-4e2b-896d-10d1470f34f0" autoplay controls loop>
</video>
<video src="https://github.com/enso-org/enso/assets/641547/a18e8bbe-3e18-4e75-b627-c0c746f73e66" autoplay controls loop>
</video>
image
image

</details>

# Scenario 3

```sh
ENSO_BUILD_TARGET=dir \
    ./run ide watch \
    --skip-wasm-opt \
    --wasm-profile=dev \
    --backend-source=release \
    --backend-release=nightly \
    --skip-version-check \
    --profiling-level=objective \
    --wasm-log-level=info \
    --wasm-uncollapsed-log-level=info \
    --ide-option=-feature-preview.new-dashboard \
    --ide-option=-debug.dev-tools \
    --ide-option=-debug \
    --ide-option=--inspect=5858 \
    --ide-option=-authentication=false
```

- ✅ `-authentication=false` runs Enso with new dashboard, but in offline mode (c.f., https://github.com/enso-org/cloud-v2/issues/481#issue-1735940816)
- ✅ User menu is completely disabled in offline mode (i.e., with `-authentication=false`), per [this comment](https://github.com/enso-org/enso/pull/6919#issuecomment-1573571013), as it's assumed that users don't want to see the authentication flow in offline mode.
- ✅ Offline mode (i.e., `-authentication=false`) should have a switcher, per [this comment](https://github.com/enso-org/enso/pull/6919#issuecomment-1582248461)
	- Addressed as implemented in [this PR](https://github.com/enso-org/enso/pull/6919) per [this comment](https://github.com/enso-org/enso/pull/6919#issuecomment-1582313488)
- ✅ Offline mode (i.e., `-authentication=false`) should have templates appear normally, occupying the full width of the page, per [this comment](https://github.com/enso-org/enso/pull/6919#issuecomment-1582248461)
	- Addressed as implemented in [this PR](https://github.com/enso-org/enso/pull/6919) per [this comment](https://github.com/enso-org/enso/pull/6919#issuecomment-1582313488)
- ✅ Logins with GitHub, Google, and password should work with offline mode (i.e., `-authentication=false`), per [this comment](https://github.com/enso-org/enso/pull/6919#issuecomment-1582313488)
- ✅ Login screen should not flash when logging in from offline mode (i.e., `-authentication=false`), per [this comment](https://github.com/enso-org/enso/pull/6919#issuecomment-1582322885)
	- Marked as TODO in the comment
	- ⚠ Wasn't able to reproduce this behaviour, but it's unclear if I missed it or if it's fixed.
- ✅ Setting username in offline mode (i.e., `-authentication=false`) should work, per [this comment](https://github.com/enso-org/enso/pull/6919#issuecomment-1582322885)
- ✅ Creating new account flow should work in offline mode (i.e., `-authentication=false`), and redirect back to the login page, per [this comment](https://github.com/enso-org/enso/pull/6919#issuecomment-1582322885)

# Scenario 4

```sh
ENSO_BUILD_TARGET=dir \
    ./run ide watch \
    --skip-wasm-opt \
    --wasm-profile=dev \
    --backend-source=release \
    --backend-release=nightly \
    --skip-version-check \
    --profiling-level=objective \
    --wasm-log-level=info \
    --wasm-uncollapsed-log-level=info \
    --ide-option=-feature-preview.new-dashboard=false \
    --ide-option=-debug.dev-tools \
    --ide-option=-debug \
    --ide-option=--inspect=5858 \
    --ide-option=-authentication
```

- ✅ `-feature-preview.new-dashboard=false` runs authentication with old dashboard; does not show a blank white page (c.f., https://github.com/enso-org/cloud-v2/issues/481#issue-1735940816)

<details>
<summary>
Screenshot
</summary>
image

</details>

# Scenario 5

```sh
ENSO_BUILD_TARGET=dir \
    ./run ide watch \
    --skip-wasm-opt \
    --wasm-profile=dev \
    --backend-source=release \
    --backend-release=nightly \
    --skip-version-check \
    --profiling-level=objective \
    --wasm-log-level=info \
    --wasm-uncollapsed-log-level=info \
    --ide-option=-feature-preview.new-dashboard=false \
    --ide-option=-debug.dev-tools \
    --ide-option=-debug \
    --ide-option=--inspect=5858 \
    --ide-option=-authentication=false
```

- ✅ `-authentication=false -feature-preview.new-dashboard=false` runs old dashboard (c.f., https://github.com/enso-org/cloud-v2/issues/481#issue-1735940816)

<details>
<summary>
Screenshot
</summary>
image

</details>

# Scenario 6

```sh
ENSO_BUILD_TARGET=dir \
    ./run ide build \
    --skip-wasm-opt \
    --wasm-profile=dev \
    --backend-source=release \
    --backend-release=nightly \
    --skip-version-check \
    --profiling-level=objective \
    --wasm-log-level=info \
    --wasm-uncollapsed-log-level=info
./dist/ide/mac-arm64/Enso.app/Contents/MacOS/Enso \
    -feature-preview.new-dashboard \
    -debug.dev-tools \
    -debug \
    --inspect=5858 \
    -authentication
```

- ✅ Nodes in Cloud project are visible
- ✅ Logs appear with `console.log("Hello")`
- ⚠️ (Re-testing) Glitchy body content resize behaviour when switching rapidly between local & cloud views (see screencasts)
    - Marked as fixed, per https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268

<details>
<summary>
Screenshot(s)/screencast(s)
</summary>
<video src="https://github.com/enso-org/enso/assets/641547/b7739ed7-573d-47b7-88e5-40411b2764f4" autoplay controls loop>
image

</video>
</details>

# Scenario 7

```sh
cd app/ide-desktop
npm run build-dashboard
```

- ✅ Command ran successfully

<details>
<summary>
Screenshot
</summary>
image

</details>

# Scenario 8

```sh
cd app/ide-desktop
npm run watch-dashboard
```

- ✅ Command ran successfully
- ✅ GUI at http://localhost:8080 opens
- ✅ IDE does not rebuild for edits in `app/gui/**/*.rs`
- ✅ IDE rebuilds for edits in `app/ide-desktop/lib/dashboard/src/authentication/src/**/*.tsx`
- ✅ IDE reloads for edits in `app/ide-desktop/lib/dashboard/src/authentication/src/**/*.tsx`
- ⚠️ (Re-testing) Nodes in Cloud project are visible
    - ~~Switcher does not appear, so no way to open cloud projects (see screenshots)~~
    - `cloud.enso.org` frontend is intended to be hosted, meaning it only has access to Cloud projects, so missing switcher is intentional (c.f., https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268)

<details>
<summary>
Screenshot
</summary>
image

</details>

**Comment by somebody1234** [comment]
- Error in logs when re-opening existing and currently open Cloud Project, Content-Length is not provided
  - :warning: unable to repro?
  - :warning: can't open cloud projects, `checkResources()` gets stuck on `400 Bad Request` for an extremely long time
- Overlap in text for file rename modal & close button for file rename modal
  - now fixed
- When a project is opened in the background, when you switch between modes, the stop icon should not be animated
  - now *should be* fixed. but :warning: can't open cloud projects, see above
- + should appear next to the project; that button should create a new project from empty template
  - duplicate toast is fixed also fixed in a later PR, but it should probably be backported here since it *is* pretty important
- When clicking the main upload file icon, no modal should appear. A finder should appear directly
  - indeed works after merging develop into this branch
- Project create form is added
  - it was changed to create a project directly - so there is no issue here
- Click avatar -> select change password -> click outside, hide modal -> click avatar (doesn't work) -> click avatar (works)
  - the issue was that no "change password" option appears. this is intentional on google and github signins, as they do not have a password (@indiv0)
- Glitchy body content resize behaviour when switching rapidly between local & cloud views (see screencasts)
  - now fixed
- Nodes in Cloud project are visible: switcher does not appear
  - `watch-dashboard` is the cloud frontend which is intended to be hosted on `cloud.enso.org` - this means it only has access to the remote (cloud) backend, so the switcher is not needed
  - :warning: to investigate later

thanks as always for the thorough PR!

**Comment by indiv0** [comment]
**TL;DR: 5 issues fixed, 2 issues remain unfixed, 1 issue fixed in future PR, 2 new issues found, QA failed ❌**

# Scenario Descriptions

<table style="border: 1px solid black;">
    <thead>
        <tr>
            <th style="border: 1px solid black;" rowspan=4>Scenario</th>
        </tr>
        <tr>
            <th style="border: 1px solid black;" colspan=5>Command</th>
            <th style="border: 1px solid black;" colspan=2>Features</th>
            <th style="border: 1px solid black;" rowspan=3>Debug Scene</th>
        </tr>
        <tr style="border: 1px solid black;">
            <th style="border: 1px solid black;">GUI</th>
            <th style="border: 1px solid black;" colspan=2>IDE</th>
            <th style="border: 1px solid black;" colspan=2>NPM</th>
            <th style="border: 1px solid black;" rowspan=2>Auth</th>
            <th style="border: 1px solid black;" rowspan=2>New Dashboard</th>
        </tr>
        <tr style="border: 1px solid black;">
            <th style="border: 1px solid black;">Watch</th>
            <th style="border: 1px solid black;">Watch</th>
            <th style="border: 1px solid black;">Build</th>
            <th style="border: 1px solid black;">build-dashboard</th>
            <th style="border: 1px solid black;">watch-dashboard</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td style="border: 1px solid black;">2</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">❌</td>
        </tr>
        <tr>
            <td style="border: 1px solid black;">8</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">✅</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
            <td style="border: 1px solid black;">-</td>
        </tr>
    </tbody>
</table>

# Scenario 2

```sh
ENSO_BUILD_TARGET=dir \
    ./run ide watch \
    --skip-wasm-opt \
    --wasm-profile=dev \
    --backend-source=local \
    --backend-release=nightly \
    --skip-version-check \
    --profiling-level=objective \
    --wasm-log-level=info \
    --wasm-uncollapsed-log-level=info \
    --ide-option=-feature-preview.new-dashboard \
    --ide-option=-debug.dev-tools \
    --ide-option=-debug \
    --ide-option=--inspect=5858 \
    --ide-option=-authentication
```

- ❌ **NEW** Intermittent 502 websocket error when opening Cloud project (see screenshots)
- ❌ **NEW** When a Cloud project fails to open because of 502 on websocket, it is no longer possible to return to the dashboard (see screencasts)
- ✅ Error in logs when re-opening existing and currently open cloud project, `Content-Length` is not provided
    - @somebody1234 was unable to reproduce (c.f., https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268)
    - Unable to reproduce, did not occur, presumed fixed
- ✅ `checkResources()` gets stuck on `400 Bad Request` for an extremely long time
    - Unable to reproduce, `checkResources()` did not get stuck for long, presumed fixed
- ❌ Overlap in text for file rename modal & close button for file rename modal
	- Not necessarily caused by this PR, but worth addressing
	- Marked as now fixed, per https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268
	- Failed during re-QA, no "Rename" option appears on files? See screenshots
- ✅ When a project is opened in the background, when you switch between modes, the `stop` icon should not be animated (c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099)
	- ~~Marked as "might have been fixed" in https://github.com/enso-org/enso/pull/6728~~
	- ~~Does not appear to be fixed (see screencasts)~~
	- Marked as now fixed, per https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268
	- No longer occurs, fixed.
- ❌ ~~`+` should appear next to the project; that button should create a new project from empty template~~
	- c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099
	- Marked as TODO per [this comment](https://github.com/enso-org/cloud-v2/issues/483#issuecomment-1571907519)
	- Tracked as https://github.com/enso-org/cloud-v2/issues/480
	- Marked as implemented in https://github.com/enso-org/enso/pull/6923
	- Icon has been added and works as described, but a duplicate "creating a new project" toast appears when it is used (see screencasts)
	- Duplicate toast is marked as fixed in a later PR (c.f., https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268), will test on that PR
- ❌ When clicking the main upload file icon, no modal should appear. A finder should appear directly. The file should be uploaded into current directory without any more confirmations (c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099)
    - Marked as TODO per [this comment](https://github.com/enso-org/cloud-v2/issues/483#issuecomment-1571907519)
    - Marked as implemented in [this PR](https://github.com/enso-org/enso/pull/6919) per [this comment](https://github.com/enso-org/enso/pull/6919#issue-1736336292)
    - ~~Not fixed; modal still appears (see screenshots)~~
        - ~~Potentially because fix hasn't been merged into this branch?~~
    - Marked as now fixed, per https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268
    - Modal still appears, see screencasts
- ✅ Click avatar -> select change password -> click outside, hide modal -> click avatar (doesn't work) -> click avatar (works)
	- c.f., https://github.com/enso-org/cloud-v2/issues/483#issue-1735981099
	- Supposed to be fixed by [Save backend type in localStorage, and automatically open projects when ready enso#6728](https://github.com/enso-org/enso/pull/6728)
	- ~~Not fixed; no "change password" option appears under the avatar menu at all (see screenshots)~~
	- No "change password" option is intentional on GitHub and Google logins, as they do not have a password (c.f., https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268)
	- Works correctly, "change password" appears outside of GitHub and Google logins, and does not appear when logging in with a federated provider
- ✅ Glitchy body content resize behaviour when switching rapidly between local & cloud views
    - Marked as fixed, per https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268
    - No longer occurs after re-test, fixed

<details>
<summary>
Screenshot(s)/screencast(s)
</summary>
image
image
<video src="https://github.com/enso-org/enso/assets/641547/dd5258b8-5de1-42b2-8392-f1001515eea6" autoplay controls loop></video>
image
<video src="https://github.com/enso-org/enso/assets/641547/78534a2c-784d-41d0-bc7d-4a652b8d594c" autoplay controls loop></video>
</details>

# Scenario 8

```sh
cd app/ide-desktop
npm run watch-dashboard
```

- ✅ Command ran successfully
- ✅ GUI at http://localhost:8080 opens
- ❌ Nodes in Cloud project are visible
    - ~~Switcher does not appear, so no way to open cloud projects~~
    - `cloud.enso.org` frontend is intended to be hosted, meaning it only has access to Cloud projects, so missing switcher is intentional (c.f., https://github.com/enso-org/enso/pull/6919#issuecomment-1596437268)
    - Unable to re-test because cloud project failed to open (see screenshots)
        - Potentially caused by same 502 websocket error reported above

<details>
<summary>
Screenshot
</summary>
image

</details>

**Comment by somebody1234** [comment]
- Intermittent 502 websocket error - can't repro but that doesn't mean this isn't an issue of course, just means I probably won't be able to investigate myself
- When a Cloud project fails to open because of 502 on websocket, it is no longer possible to return to the dashboard - this is an IDE error, we listen to the event. Reported as #7077
- Overlap in text for file rename modal & close button for file rename modal
  - You might have meant project rename - I don't remember files and secrets ever supporting rename (because there is no corresponding backend endpoint)
  - Double-checked that the close button doesn't overlap on project rename and project delete modals
- When clicking the main upload file icon, no modal should appear.
  - I fixed the main upload file icon, but forgot about the one in the files table. now fixed

**Comment by PabloBuchu** [comment]
@somebody1234 the templates acts very weird when switching between modes.. can we somehow minimize such bugs? please do a self qa your PRs

https://github.com/enso-org/enso/assets/2855109/67cc1f0e-a725-4bd8-b33b-4c07104439fa

**Comment by somebody1234** [comment]
@PabloBuchu That is mostly intentional - I changed the templates to have an automatic number of columns - so when one view has a scrollbar and the other doesn't, one may have <n> columns and the other has <n+1>.

Specifically, I tested at the intended width (1280px), which shows 5 columns regardless of whether there is a scrollbar, but it is true that having that happen at any width isn't ideal
